media/js/mozorg/home/takeover.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+$(function () {
+    'use strict';
+
+    var $takeoverModal = $('#fundraising_takeover');
+    var $overlay = $('.page-overlay');
+    var $closeModal = $('#close_takeover');
+    var $form = $('#fundraiser-form');
+
+    var $donateOptionsSection = $('.donate-options');
+    var $donateCustom = $('#donate-custom');
+    var $optionLabels = $('label[class!="own-amount"]', $donateOptionsSection);
+    var $options = $('input[type="radio"]', $donateOptionsSection);
+
+    // sets the mozorg.takeover state to true indicating that the user has
+    // seen the takeover modal during their current session.
+    function storeSession() {
+        try {
+            sessionStorage.setItem('mozorg.takeover.seen', true);
+        } catch (ex) {
+            console.log('Error thrown while setting takeover modal state in sessionStorage: ', ex);
+        }
+    }
+
+    // Show and hides the takeover modal
+    function toggleModal() {
+        $([$takeoverModal, $overlay]).each(function() {
+            $(this).toggleClass('hidden');
+        });
+    }
+
+    // removes checked from all the radio buttons
+    // and selected from the labels.
+    function clearOptions() {
+        $optionLabels.removeClass('selected');
+        $options.removeAttr('checked');
+    }
+
+    // handles clicks on the labels of the radio input elements.
+    $optionLabels.on('click', function() {
+        // clear selected class from labels
+        $optionLabels.removeClass('selected');
+        // empty any value that might have been typed
+        // into the custom amount field.
+        $donateCustom.val('');
+        // set the clicked label to selected
+        $(this).addClass('selected');
+    });
+
+    // once a user types into the custom amount field
+    // clear all other options.
+    $donateCustom.keyup(clearOptions);
+
+    // the name of the parameter that indicates the amount the user
+    // wish to donate has to be called amount. Seeing that we have
+    // radio buttons with that name as well as an input field, we need
+    // to disable the input field prior to form submission to avoid a
+    // scenario where we have ?amount=50&amount=
+    $form.on('submit', function() {
+        if ($donateCustom.val() === '') {
+            $donateCustom.attr('disabled', 'disabled');
+        }
+    });
+
+    // if a user clicks on 'continue to mozilla.org',
+    // close the modal and overlay
+    $closeModal.on('click', function(event) {
+        event.preventDefault();
+        toggleModal();
+    });
+
+    try {
```

Comment:
Seeing as this promo is waffled, should we add a check here for `length` to make sure it exists in the DOM before continuing?